drive/drive-space.go

```diff
@@ -0,0 +1,323 @@
+package drive
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	fxiface "github.com/functionland/go-fula/fxfs/core/iface"
+	"github.com/functionland/go-fula/fxfs/core/pfs"
+	files "github.com/ipfs/go-ipfs-files"
+	ipfsifsce "github.com/ipfs/interface-go-ipfs-core"
+	"github.com/ipfs/interface-go-ipfs-core/path"
+)
+
+type DRIVE_SPACE_TYPE string
+
+const (
+	PUBLIC_DRIVE_SPACE_TYPE  DRIVE_SPACE_TYPE = "public"
+	PRIVATE_DRIVE_SPACE_TYPE DRIVE_SPACE_TYPE = "private"
+)
+
+type MkDirOpts struct {
+	recursive bool
+}
+
+type WriteFileOpts struct {
+	parents   bool
+	overwrite bool
+}
+
+type ReadFileOpts struct {
+}
+
+type DeleteFileOpts struct {
+}
+
+type ListEntriesOpts struct {
+}
+
+type ListEntry struct {
+	ipfsifsce.DirEntry
+}
+
+type DriveSpace struct {
+	ctx       context.Context
+	api       fxiface.CoreAPI
+	SpaceType DRIVE_SPACE_TYPE
+	rootCid   string
+	RootDir   files.Directory
+}
+
+type DrivePublicSpace struct {
+	DriveSpace
+}
+
+type DrivePrivateSpace struct {
+	DriveSpace
+}
+
+// MkDir creates a new directory inside a DriveSpace given a path
+func (ds *DriveSpace) MkDir(p string, options MkDirOpts) (string, error) {
```

Comment:
The options are exported but not their fields. Do we need to define option function as well and take it as varargs instead of structs? Otherwise, as is the options are not configurable outside of this package. Same for all the other options structs.

Reply:
We have to take a deep look into the options for each action. no option is used in the implementation yet, but we will need them. After defining them, we can think about the interface. I removed the unused option (`recursive`) for now